pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,320 @@
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+					Volumes:            volumes,
+				},
+			},
+		},
+	}
+}
+
+func (c *Component) roleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     GoldmaneRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      GoldmaneRoleName,
+				Namespace: GoldmaneNamespace,
+			},
+		},
+	}
+}
+
+func (c *Component) role() *rbacv1.Role {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "patch"},
+		},
+	}
+
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		Rules: policyRules,
+	}
+}
+
+func (c *Component) deploymentSelector() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app.kubernetes.io/name": GoldmaneDeploymentName,
+		},
+	}
+}
+
+func (c *Component) networkPolicy() *netv1.NetworkPolicy {
+	return &netv1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "networking.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "allow-goldmane", Namespace: GoldmaneNamespace},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: *c.deploymentSelector(),
+			PolicyTypes: []netv1.PolicyType{netv1.PolicyTypeIngress, netv1.PolicyTypeEgress},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				{
+					Ports: []netv1.NetworkPolicyPort{{
+						Protocol: ptr.ToPtr(corev1.ProtocolTCP),
+						Port:     ptr.ToPtr(intstr.FromInt32(GoldmaneServicePort)),
+					}},
+				},
+			},
+			Egress: []netv1.NetworkPolicyEgressRule{
+				{
+					To: []netv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"app.kubernetes.io/name": GuardianDeploymentName,
+								},
+							},
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"projectcalico.org/name": GuardianNamespace,
+								},
+							},
+						},
+					},
+					Ports: []netv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.ToPtr(corev1.ProtocolTCP),
+							Port:     ptr.ToPtr(intstr.FromInt32(GuardianTargetPort)),
+						},
+					},
+				},
+				{
+					Ports: []netv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.ToPtr(corev1.ProtocolUDP),
+							Port:     ptr.ToPtr(intstr.FromInt32(53)),
+						},
+						{
+							Protocol: ptr.ToPtr(corev1.ProtocolTCP),
```

Comment:
What is this port for? 

In general I think it's good practice to add a comment to each rule explaining its intended purpose, for posterity.

Reply:
k8s API I'll add a comment.

Reply:
I think a port won't be enough in this case necessarily, since the API server can run on different ports?

You may want to use a Calico network policy, which would allow using a service selector like we do elsewhere: https://github.com/tigera/operator/blob/cd83253b701673b4536c0d1190a134593cde67eb/pkg/render/common/networkpolicy/networkpolicy.go#L204-L209

Reply:
Is this compatible? More wondering why aren't we using a calico net policy for the apiserver?

I suppose I could always watch the k8s service and pull the port from there to add it (might prefer that if you don't mind).

Reply:
Chatted offline, we're taking out the egress rules altogether because it's a bit more complex to actually start using v3 policies since we can't rely on the api server being there in calico.